script.js

```diff
@@ -166,4 +316,54 @@ function addIngredient() {
   ingredientsContainer.appendChild(newDiv);
 }
 
-addRecipeToPage(recipes[0]);
+// //**************** Search betwween recipes *****************/
+function searchClicked() {
+  const searchTextBox = document.getElementById("search");
+  const searchTextBoxValue = searchTextBox.value.toLowerCase();
+  const recipesContainer = document.getElementById("cards-container");
+  recipesContainer.innerHTML = "";
+
+  const searchResult = recipes.filter((recipe) =>
+    recipe.title.toLowerCase().includes(searchTextBoxValue)
+  );
+  searchResult.forEach((item) => {
+    addRecipeToGrid(item);
+  });
+}
+//**************** Sort Ingredients *****************/
+
+function sortAscClicked() {
+  recipes.sort(function (recipeA, recipeB) {
+    if (recipeA.ingredients.length > recipeB.ingredients.length) {
+      return 1;
+    }
+    if (recipeA.ingredients.length < recipeB.ingredients.length) {
+      return -1;
+    }
+    return 0;
+  });
+  const recipesContainer = document.getElementById("cards-container");
+  recipesContainer.innerHTML = "";
+  recipes.forEach((item) => {
+    addRecipeToGrid(item);
+  });
+}
+
+function sortDesClicked() {
+  recipes.sort(function (recipeA, recipeB) {
+    if (recipeA.ingredients.length < recipeB.ingredients.length) {
+      return 1;
+    }
+    if (recipeA.ingredients.length > recipeB.ingredients.length) {
+      return -1;
+    }
+    return 0;
+  });
+  const recipesContainer = document.getElementById("cards-container");
+  recipesContainer.innerHTML = "";
+
+  recipes.forEach((item) => {
+    addRecipeToGrid(item);
+  });
+}
```

Comment:
Lots of repetition here. You can split some of the logic to a new function and then have to separate ones for Des and Asc where you pass property for the order instead:

```suggestion
function sortRecipes(order) {
  const recipesContainer = document.getElementById("cards-container");
  recipesContainer.innerHTML = "";

  recipes.sort(function (recipeA, recipeB) {
    if (order === "asc") {
      return recipeA.ingredients.length - recipeB.ingredients.length;
    } else {
      return recipeB.ingredients.length - recipeA.ingredients.length;
    }
  });

  recipes.forEach((item) => {
    addRecipeToGrid(item);
  });
}

function sortAscClicked() {
  sortRecipes("asc");
}

function sortDesClicked() {
  sortRecipes("desc");
}
```